HARP-2023-Summer/Mapping/WSP_Regional_Summaries.Rmd

```diff
@@ -726,13 +649,24 @@ rseg_no_geom <- st_drop_geometry(rsegs) #copy of rsegs
 
   # base info
 for (k in 1:length(run_config$riverseg_metrics)){
-  rseg_display <- rseg_no_geom # must reinitialize rseg_display each time since we remove unused columns below
+  
+  data_set <- run_config$riverseg_metrics[[k]]$data_set
+  
+  if (data_set == "rseg") {
```

Comment:
@EllaF21 I think we need to only set the variable `rseg_display` here in this if statement, since you can just as easily assign `run_config$riverseg_metrics[[k]]$column_name <- NA` in the `data_set` config. 

Also, as an alternative to the `if-then` construct with predetermined options for 'rseg' and 'facility', you could indicate *any* dataframe by name in `data_set` and use the `get()` function to return that to rseg_display, or throw an error and/or quit the function.  Like so:
```
table_cols <- unlist(run_config$riverseg_metrics[[k]]$tables_cols_r)
display_column <- run_config$riverseg_metrics[[k]]$column_name
# test for existence of data_set variable (i.e. data_set = "rseg_no_geom", or data_set = "faclis_nogeom"
if (exists(data_set)) {
  rseg_display <- get(data_set)
} else {
  message(paste("The requested data_set,", data_set,"is not defined. Please adjust settings in config file")
}
```

With the above syntax we could be very flexible in the future if we had other table we wanted to pass in.  This is *not* a necessity, the simple if-then to set just the table would be fine:
```
if (data_set == 'facil') {
  rseg_display <- facils_nogeom
} else {
  rseg_display <- rseg_nogeom
}
```